### DISCUSSION TITLE
Temporary file cleanup not working on s3

### DISCUSSION THREAD

**Discussion by thierroboto** [livewire/livewire#8543] [1. Help]
Today I looked at my s3 bucket and saw that my livewire-tmp folder is full of files that are months old. I did originally do php artisan livewire:configure-s3-upload-cleanup. Is there anything i'm missing? Is there a job i can run so that it makes sure the files are deleted? Thank-you.

**Comment by puklipo** [comment]
I noticed that the prefix of the S3 lifecycle rule had changed from livewire-tmp to tmp/.

1. Delete the existing rule
2. Run `php artisan livewire:configure-s3-upload-cleanup`
3. Confirm that the rule prefix is `​​livewire-tmp`
4. Then, a few hours or days later, it changed to `tmp/`.

The only solution might be to publish `config/livewire.php` and change it to `tmp/`?

```php
'directory' => 'tmp',   // Example: 'tmp'                      | Default: 'livewire-tmp'
```

**Comment by puklipo** [comment]
I noticed something else.

This is caused by Laravel Vapor.

1. Livewire: Create new rule. prefix `livewire-tmp`
2. Vapor: When deploying, delete all rules. Create new rule. prefix `tmp/`

The solution is the same.
```php
'directory' => 'tmp',
```

**Reply by thierroboto** [reply]
I am indeed on laravel vapor. Thanks for the help! I'll try this and report back.

**Comment by ken717w** [comment]
I just run into this issue too.

Running `php artisan livewire:configure-s3-upload-cleanup` gives `Failed to configure S3 bucket [my-bucket] to automatically cleanup files older than 24hrs!`.

At the beginning I didn't even have `config/livewire.php`, so I created it by running `php artisan livewire:publish --config` and then tried to setup auto cleanup again, but still failed.

**Reply by alexcavender** [reply]
Hi @ken717w Did you ever get this resolved? I'm running into similar issue. Works fine on MinIO locally, but is failing with a live bucket

**Reply by ken717w** [reply]
No, the impact of this is relatively small, so didn't spend too much time on it.

**Comment by jordyulburghs** [comment]
Hi i ran in to similar problems using laravel cloud and got a response that might be helpfull from their staff.
They say this : 

This happens because Cloudflare R2 doesn’t fully support the PutBucketLifecycleConfiguration API that Livewire tries to use for automatic cleanup. That’s why you’re seeing the MalformedXML response—it’s not an issue with Laravel Cloud itself, just a difference in how R2 handles lifecycle configuration.

They adviced to setup a custom cleanup process not relying on livewire.

**Reply by grpaiva** [reply]
I was looking for exactly this. I'm also using Laravel Cloud and will need some sort of automatic cleanup as well. 

Thanks for sharing!
